src/WTG.Analyzers.Test/TestData/VisibilityAnalyzer/Partial/Diagnostics.xml

```diff
@@ -1,10 +1,17 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <diagnostics severity="Hidden">
 	<languageVersion>9.0</languageVersion>
+	<suppressId justtification="To ignore when only one method in one class have the code fix removing 'private' applied.">CS8799</suppressId>
```

Comment:
I don't quite follow what this is saying, can we avoid the warning/error entirely?

Also there is a typo in the attribute.

Reply:
In this partial members unittest,

1. reads `Source.cs` file
2. get diagnostics to fix
3. fix one then check any further violation
    3.1. if no-new violation, proceeds to next fix
    3.2. if any, fails

![image](https://github.com/WiseTechGlobal/WTG.Analyzers/assets/173008292/75ca729c-2b1d-4c80-832f-c2a0b92a9f28)

In here, it removes `private` from 1 method above, then check additional violation, then, it triggers `CS8799: Both partial method declarations must have identical accessibility modifiers.` in the middle, before removing `private` from 2.

That's the reason why I added `suppressId CS8799` in the xml.

I wanted to explain this in `justification` attribute (not a thing in this analyzer though). I may need to add `<!-- -->` xml comment instead.

Another approach I can think of here is that the additional violation checking logic can be moved out of loop, so that the checking can be performed after all fixes in a unittest are done.
e.g. the checking can be done after 1, 2, 3 and 4 code fixes done in the above example.

Reply:
This might be a sign that you need to write a custom Bulk-Fixer, but I'm not 100% sure. Perhaps @brian-reichle can advise here?

Reply:
The bulk fixer is for when fixing results individually produces results that cannot merge together or where fixing one affects the other. In this case it's because the partial declaration and partial implementation should be fixed together as a single unit.

I suppose the ideal outcome would be to only report this on the partial declaration and then have the code fix also update the corresponding implementations; though detecting and matching the implementations to the declarations might be more trouble than it's worth, in which case suppressing CS8799 would be the next-best option.

Reply:
Are there any real-world cases where we have to supply the `private` keyword because a Source Generator emits the other half and adds the `private` keyword there?

Reply:
Not that I can think of. `EF`, `Blazor`, `Winforms` (as in .Designer.cs), I haven't seen from them. I'm not sure others that may emit `private` in default.

Reply:
Can you check the Regex and Logging source generators? Those are now provided in-box by the .NET SDK IIRC.

Reply:
If I understand correctly, I suppose this is what you mean?
<img width="384" alt="image" src="https://github.com/WiseTechGlobal/WTG.Analyzers/assets/173008292/f2fe0261-89f7-4a23-8a98-afcd848d4a0f">
I couldn't find any in them. However, using `<EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>` in `Blazor` app and checked the generated files from the designated folder and couldn't find any `private` keyword.

Also, I tried to create samples with `GeneratedRegex` and `LoggerMessage` attribute to generate files in those generators and they generate the same modifier used in original files, which means the original already checked/fixed by our Analyzer as far as I understand.
<img width="1210" alt="image" src="https://github.com/WiseTechGlobal/WTG.Analyzers/assets/173008292/41c44716-aeb5-488f-901d-775dbc0ac52e">